### PR TITLE
fix: fix memory growth tests

### PR DESCRIPTION
**Pull request by 0xVolosnikov**
# What ❔

Fixes memory growth assembly tests; adds new memory growth assembly tests

## Why ❔

Current tests are broken and don't check anything

Fixes EVM-655

### CODE REVIEW

File: src/tests/simple_tests/asm_tests.rs [hunk 1]
```diff
@@ -11,25 +11,29 @@ use zkevm_assembly::Assembly;
...
 /// contracts should be in `ADDRESS.asm` files, where `ADDRESS` is the numerical
 /// address at which they should be deployed.
-pub fn run_asm_based_test(test_dir: &str, additional_contracts: &[i32], options: Options) {
-    run_asm_based_test_template(test_dir, additional_contracts, options, None);
+pub fn run_asm_based_test(
+    test_dir: &str,
+    additional_contracts_addresses: &[i32],
+    options: Options,
+) {
+    let mut additional_contracts = Vec::<(String, i32)>::new();
```
Comment (mm-zk) [inline src/tests/simple_tests/asm_tests.rs]: nit: you could do it as 'iter' with map.

File: src/tests/simple_tests/context.rs [hunk 2]
```diff
@@ -1,31 +1,12 @@
 use super::*;
+use crate::tests::simple_tests::asm_tests::run_asm_based_test;
 
+// TODO meta isn't checked
```
Comment (mm-zk) [inline src/tests/simple_tests/context.rs]: can you explain more? 
(and if needed, create a ticket and reference it here)

**Review by mm-zk — APPROVED** [review]

File: src/tests/simple_tests/context.rs (hunk 2 above)
Comment (0xVolosnikov) [reply src/tests/simple_tests/context.rs]: We do not check values returned by meta opcode in this test

Comment (0xVolosnikov) [reply src/tests/simple_tests/context.rs]: But I'm not sure that we want to create separate test cases for it

Comment (0xVolosnikov) [reply src/tests/simple_tests/context.rs]: EVM-659

**Pull request merged**